lib/mkCrossToolchainEnv.nix

```diff
@@ -0,0 +1,54 @@
+{ pkgs
+, lib
+,
+}: stdenvSelector:
+let
+  nativePkgs = pkgs.pkgsBuildBuild;
+
+  hostStdenv = stdenvSelector pkgs.pkgsBuildHost;
+  targetStdenv = stdenvSelector pkgs.pkgsHostTarget;
+
+  varsForPlatform = buildKind: stdenv:
+    let
+      ccPrefix = stdenv.cc.targetPrefix;
+      cargoEnv = stdenv.hostPlatform.rust.cargoEnvVarTarget;
+    in
+    {
+      # Point cargo to the correct linker
+      "CARGO_TARGET_${cargoEnv}_LINKER" = "${ccPrefix}cc";
+
+      # Set environment variables for the cc crate (see https://docs.rs/cc/latest/cc/#external-configuration-via-environment-variables)
+      "CC_${cargoEnv}" = "${ccPrefix}cc";
+      "CXX_${cargoEnv}" = "${ccPrefix}c++";
+      "AR_${cargoEnv}" = "${ccPrefix}ar";
+
+      # Set environment variables for the cc crate again, this time using the build kind
+      # In theory, this should be redundant since we already set their equivalents above, but we set them again just to be sure
+      # This way other potential users of e.g. "HOST_CC" also use the correct toolchain
+      "${buildKind}_CC" = "${ccPrefix}cc";
+      "${buildKind}_CXX" = "${ccPrefix}c++";
+      "${buildKind}_AR" = "${ccPrefix}ar";
+    }
+    # Configure an emulator for the platform (if we need one, and there's one available)
+    // (lib.optionalAttrs (!(stdenv.buildPlatform.canExecute stdenv.hostPlatform) && stdenv.hostPlatform.emulatorAvailable nativePkgs) {
+      "CARGO_TARGET_${cargoEnv}_RUNNER" = stdenv.hostPlatform.emulator nativePkgs;
+    });
+in
+if pkgs.buildPlatform == pkgs.hostPlatform
+then { }
+else
```

Comment:
Could be simplified to `lib.optionalAttrs (pkgs.buildPlatform != pkgs.hostPlatform) (lib.mergeAttrsList [ ... ])`

Reply:
Resolved in latest push.